README.md

```diff
@@ -311,11 +311,13 @@ These are the available config options for making requests. Only the `url` is re
   // If set to 0, no redirects will be followed.
   maxRedirects: 5, // default
 
-  // `httpAgent` and `httpsAgent` define a custom agent to be used when performing http
-  // and https requests, respectively, in node.js. This allows options to be added like
-  // `keepAlive` that are not enabled by default.
-  httpAgent: new http.Agent({ keepAlive: true }),
-  httpsAgent: new https.Agent({ keepAlive: true }),
+  // `agents` define custom agents to be used when performing http
+  // and https requests in node.js.
+  // This option allows features like keep alive requests.
+  agents: {
```

Comment:
@rubennorte Do we really need to make this breaking change to the API? I think the current API is good enough and should not be changed without a strong reason.

Reply:
Excuse me. How about this fixing in adapters/http.js.
```
if (config.maxRedirects) {
  options.maxRedirects = config.maxRedirects;
}
options.agents = {
  http: config.httpAgent,
  https: config.httpsAgent
};
transport = isHttpsProxy ? httpsFollow : httpFollow;
```
https://github.com/axios/axios/blob/master/lib/adapters/http.js#L169-L172